docs/services/inject-async.service.ts

```diff
@@ -0,0 +1,94 @@
+/*!
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.dev/license
+ */
+
+import {
+  DestroyRef,
+  ENVIRONMENT_INITIALIZER,
+  EnvironmentInjector,
+  Injectable,
+  Injector,
+  Provider,
+  ProviderToken,
+  Type,
+  createEnvironmentInjector,
+  inject,
+} from '@angular/core';
+
+/**
+ * inject a service asynchronously
```

Comment:
optional: it would be nice to have a better comment on what this service actually does

(it seems like it is mostly a convenience for lazy-loading the injection token)